search.go

```diff
@@ -0,0 +1,66 @@
+package parquet
+
+// Search uses the column index passed as argument to find the page that the
+// given value is expected to be found in.
+//
+// The function returns the index of the first page that might contain the
+// value. If the function determines that the value does not exist in the
+// index, NumPages is returned.
+//
+// The comparison function passed as last argument is used to determine the
+// relative order of values. This should generally be the Compare method of
+// the column type.
+func Search(index ColumnIndex, value Value, cmp func(Value, Value) int) int {
```

Comment:
I am not intimately familiar with parquet but I think it might be possible that more than one page can contain the same value. If so, should this return multiple page indices?

Additionally, will the name of this function cause conflicts with other types of searches, like searching within a page to find matches of row indices or values?

Reply:
The intent would be for the application to start reading pages from the first and forward, assuming values are ordered, it can stop of the first page that doesn't contain the value it's looking for. If values are not ordered, it needs to scan everything either ways, so no real benefits of returning a list of all page indexes in that case.